kbase-extension/static/kbase/js/common/jobManager.js

```diff
@@ -788,12 +788,28 @@ define([
 
             _initJobs(args) {
                 const { batchId } = args;
+                if (!batchId) {
+                    throw new Error('Cannot init jobs without a batch ID');
+                }
+
+                const self = this;
                 ['ERROR', 'INFO', 'LOGS', 'RETRY', 'STATUS'].forEach((msgType) => {
                     this.addListener(jcm.MESSAGE_TYPE[msgType], jcm.CHANNEL.BATCH, batchId);
                 });
 
-                // request job updates
-                this.bus.emit(jcm.MESSAGE_TYPE.START_UPDATE, { [jcm.PARAM.BATCH_ID]: batchId });
+                if (!this.looper) {
+                    this.looper = new Looper();
+                }
+
+                // set up repeated requests for batch status
+                // on receiving a job status message, this handler will set
+                // up a delayed request to request a status update for the batch.
+                this.addEventHandler(jcm.MESSAGE_TYPE.STATUS, {
+                    batchStatus: () => {
+                        self.looper.scheduleRequest(self.requestBatchStatus.bind(self));
+                    },
+                });
```

Comment:
possible case to consider: what happens if there is no response from the backend? Should there be a fallback where the job manager sends another request _n_ seconds after the first one if there is no response?

Reply:
Maybe? There's also a case where there are a zillion running jobs, so it just takes longer than expected for the backend to return info. So a fallback loop might clog the queue. Maybe it should inspect the kernel to see if it's busy before firing again?